<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />

  <arg name="simulation" default="true" />

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="sensor_neighbor_manager" />
  <arg name="n_threads" default="8" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <node pkg="nodelet" type="nodelet" name="sensor_neighbor" args="$(arg nodelet) sensor_neighbor/SensorNeighbor $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">
	<!-- ROS parameters config file -->	
	<rosparam file="$(find flocking)/config/uav_names.yaml" />
	<rosparam file="$(find flocking)/config/experiment.yaml" />
  </node>

</launch>
